test/linalg/tridiag.jl

```diff
@@ -348,109 +198,114 @@ let n = 12 #Size of matrix problem to test
+                    @test_throws DimensionMismatch Base.LinAlg.A_mul_B!(zeros(fA),A,ones(elty,n+1,n))
                     @test_throws DimensionMismatch A_mul_B!(zeros(elty,n,n),B,ones(elty,n+1,n))
                     @test_throws DimensionMismatch A_mul_B!(zeros(elty,n+1,n),B,ones(elty,n,n))
                     @test_throws DimensionMismatch A_mul_B!(zeros(elty,n,n+1),B,ones(elty,n,n))
                 end
             end
-        end
-    end
-    @testset "Tridiagonal matrices" begin
-        for relty in (Float32, Float64), elty in (relty, Complex{relty})
-            a = convert(Vector{elty}, randn(n - 1))
-            b = convert(Vector{elty}, randn(n))
-            c = convert(Vector{elty}, randn(n - 1))
-            if elty <: Complex
-                a += im*convert(Vector{elty}, randn(n - 1))
-                b += im*convert(Vector{elty}, randn(n))
-                c += im*convert(Vector{elty}, randn(n - 1))
-            end
-
-            @test_throws ArgumentError Tridiagonal(a,a,a)
-            A = Tridiagonal(a, b, c)
-            fA = map(elty <: Complex ? Complex128 : Float64, Array(A))
-
-            @testset "similar, size, and copy!" begin
-                B = similar(A)
-                @test size(B) == size(A)
-                copy!(B,A)
-                @test B == A
-                @test isa(similar(A), Tridiagonal{elty})
-                @test isa(similar(A, Int), Tridiagonal{Int})
-                @test isa(similar(A, Int, (3,2)), Matrix{Int})
-                @test size(A,3) == 1
-                @test_throws ArgumentError size(A,0)
-            end
-            @testset "Diagonal extraction" begin
-                @test diag(A,-1) == a
-                @test diag(A,0) == b
-                @test diag(A,1) == c
-                @test diag(A,n-1) == zeros(elty,1)
-                @test_throws ArgumentError diag(A,n+1)
-            end
-            @testset "Simple unary functions" begin
-                for func in (det, inv)
-                    @test func(A) ≈ func(fA) atol=n^2*sqrt(eps(relty))
+            if mat_type == SymTridiagonal
+                @testset "Tridiagonal/SymTridiagonal mixing ops" begin
+                    B = convert(Tridiagonal{elty}, A)
+                    @test B == A
+                    @test B + A == A + B
+                    @test B - A == A - B
                 end
-            end
-            if elty <: Real
-                @testset "Rounding to Ints" begin
-                    @test round.(Int,A) == round.(Int,fA)
-                    @test isa(round.(Int,A), Tridiagonal)
-                    @test trunc.(Int,A) == trunc.(Int,fA)
-                    @test isa(trunc.(Int,A), Tridiagonal)
-                    @test ceil.(Int,A) == ceil.(Int,fA)
-                    @test isa(ceil.(Int,A), Tridiagonal)
-                    @test floor.(Int,A) == floor.(Int,fA)
-                    @test isa(floor.(Int,A), Tridiagonal)
+                if elty <: Real
```

Comment:
Seems like these two `if`s could be collapsed to one level with `if elty <: LinAlg.BlasReal`